src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        int salary;
+
+        result.append(String.format("Report for period %s - %s", dateFrom, dateTo))
+                .append(System.lineSeparator());
+        for (int i = 0; i < names.length; i++) {
+            salary = 0;
+            for (String nameFromData : data) {
+
+                if (names[i].equals(nameFromData.split(" ")[1])
```

Comment:
create a String array variable and reuse it instead of doing split every single time